ocaml/database/redo_log.ml

```diff
@@ -542,7 +544,10 @@ let maybe_retry f log =
 (* Functions relating to the lifecycle of the block device I/O process. *)
 
 (* Close any existing socket and kill the corresponding process. *)
-let shutdown log =
+
+let shutdown_m = Mutex.create ()
```

Comment:
It seems to me that the resource in this case is `!(log.pid)`, having a mutex to be able to access it would make more sense that one for just the shutdown